include/deal.II/distributed/tria.h

```diff
@@ -880,7 +881,37 @@ namespace parallel
         std::map<unsigned int, pack_callback_t> pack_callbacks;
       };
 
-      CellAttachedData cell_attached_data;
+      /**
+       * Vector of tuples containing of p4est quadrant, dealii cells and their relation after refinement.
+       * This makes a one to one correlation possible and allows for parallelization in the future.
+       *
+       * To write its contents, call set_quadrant_cell_pairs(). This vector will be ordered by the occurance of
+       * quadrants in the corresponding p4est sc_array., at which data for transfer will be packed.
+       */
+      std::vector< std::tuple< typename dealii::internal::p4est::types<dim>::quadrant *,
+          typename Triangulation<dim,spacedim>::CellStatus,
+          typename Triangulation<dim,spacedim>::cell_iterator> > local_quadrant_cell_relations;
```

Comment:
so question: is this a data structure that we need frequently and therefore need to store, or does it need to be regenerated for each use anyway? If the former, then it makes sense to keep it as a member variable. If we don't, then it may be simpler to just build it as a local variable in the function where it is used.